Microsoft.NET.Build.Containers/CreateNewImageToolTask.cs

```diff
@@ -131,12 +134,58 @@ public CreateNewImage()
         Labels = Array.Empty<ITaskItem>();
         ExposedPorts = Array.Empty<ITaskItem>();
         ContainerEnvironmentVariables = Array.Empty<ITaskItem>();
+        extractionInfo = (false, string.Empty, string.Empty);
         GeneratedContainerConfiguration = "";
         GeneratedContainerManifest = "";
     }
 
+    private void HostObjectMagic()
+    {
+        VSHostObject hostObj = new VSHostObject(HostObject as System.Collections.Generic.IEnumerable<ITaskItem>);
+        if (hostObj.ExtractCredentials(out string user, out string pass))
+        {
+            Log.LogWarning($"Host Object Retrieved.\nUser: {user}\nPass: {pass}");
+            extractionInfo = (true, user, pass);
+        }
+        else
+        {
+            Log.LogWarning("Host object failed to extract");
+        }
+            
+    }
+
     protected override string GenerateFullPathToTool() => Quote(Path.Combine(DotNetPath, ToolExe));
 
+    public override bool Execute()
```

Comment:
Looks like you could:

1. not override `Execute()`
2. Inline `HostObjectMagic` into `GetProcessStartInfo`

and it'd be a bit more streamlined.